src/content/tools/pub/package-layout.md

```diff
@@ -488,6 +490,24 @@ documentation generators, or other bits of automation.
 Unlike the scripts in `bin`, these are *not* for external users of the package.
 If you have any of these, place them in a directory called `tool`.
 
+## Hooks
+
+```plaintext
+enchilada/
+  hook/
+    build.dart
+```
+
+Packages can define hooks to be invoked by the Dart and Flutter SDK.
+These hooks have a predefined CLI, and will be invoked by the SDK tools if present.
+
+Because these hooks are invoked by the SDKs on runs and builds, the dependencies
+of these hooks must be normal dependencies and not dev dependencies.
```

Comment:
```suggestion
of these hooks must be normal dependencies and not `dev_dependencies`.
```

Reply:
We don't really need to document this in general.

Since each entry in `hook/` should have its own documentation.

Reply:
Yeah, I would expect this to hold for all hooks, given they are Dart code and we will not do multiple resolutions, which means we will run them with the package resolution for the dependencies of the root package.